controllers/components/kserve/kserve_controller.go

```diff
@@ -19,40 +19,99 @@ package kserve
-	Scheme *runtime.Scheme
-}
+// NewComponentReconciler creates a ComponentReconciler for the Dashboard API.
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(mgr, componentsv1.KserveInstanceName, &componentsv1.Kserve{}).
+		// operands - owned
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}, builder.WithPredicates(clusterRoleAggregationRulePredicate)).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		// The ovms template gets a new resourceVersion periodically without any other
+		// changes. The compareHashPredicate ensures that we don't needlessly enqueue
+		// requests if there are no changes that we don't care about.
+		Owns(&templatev1.Template{}, builder.WithPredicates(compareHashPredicate)).
+		Owns(&featuresv1.FeatureTracker{}).
+		Owns(&networkingv1.NetworkPolicy{}).
+		Owns(&monitoringv1.ServiceMonitor{}).
+		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
+		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		Owns(&appsv1.Deployment{}, builder.WithPredicates(resources.NewDeploymentPredicate())).
+		// operands - watched
+		//
+		// By default the Watches functions adds:
+		// - an event handler mapping to a cluster scope resource identified by the
+		//   components.opendatahub.io/managed-by annotation
+		// - a predicate that check for generation change for Delete/Updates events
+		//   for to objects that have the label components.opendatahub.io/managed-by
+		//   set to the current owner
+		//
+		Watches(&extv1.CustomResourceDefinition{}).
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the Kserve object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *KserveReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+		// TODO dynamic watch these if kinds exist?
+		// See: https://issues.redhat.com/browse/RHOAIENG-15170
+		// WatchesGVK(gvk.KnativeServing, enqueueRequestsForChildFT(mgr.GetClient())).
+		// WatchesGVK(gvk.ServiceMeshMember, enqueueRequestsForChildFT(mgr.GetClient())).
+		// WatchesGVK(gvk.EnvoyFilter, enqueueRequestsForChildFT(mgr.GetClient())).
+		// WatchesGVK(gvk.AuthorizationPolicy, enqueueRequestsForChildFT(mgr.GetClient())).
+		// WatchesGVK(gvk.Gateway, enqueueRequestsForChildFT(mgr.GetClient())).
```

Comment:
FYI, https://github.com/opendatahub-io/opendatahub-operator/pull/1385 has been merged so it is now possible to add dynamic watches